### ISSUE TITLE
EAS undefined __apply_Xcode_12_5_M1_post_install_workaround

### ISSUE THREAD

**Issue by santomegonzalo** [expo/eas-cli#2168]
### Summary

Using EAS and the latest version of ReactNative will break iOS build. This is because react-native 0.73 removed the call `__apply_Xcode_12_5_M1_post_install_workaround`.

### Managed or bare?

Managed

### Environment

Issues:
  Some dependencies are incompatible with the installed expo version:
    @react-native-async-storage/async-storage@1.21.0 - expected version: 1.18.2
    @react-native-community/datetimepicker@7.6.2 - expected version: 7.2.0
    @react-native-community/netinfo@11.2.1 - expected version: 9.3.10
    @sentry/react-native@5.15.2 - expected version: 5.10.0
    @shopify/flash-list@1.6.3 - expected version: 1.4.3
    expo-application@5.4.0 - expected version: ~5.3.0
    expo-clipboard@4.5.0 - expected version: ~4.3.1
    expo-device@5.6.0 - expected version: ~5.4.0
    expo-font@11.6.0 - expected version: ~11.4.0
    expo-haptics@12.6.0 - expected version: ~12.4.0
    expo-image@1.5.1 - expected version: ~1.3.5
    expo-linear-gradient@12.5.0 - expected version: ~12.3.0
    expo-linking@6.0.0 - expected version: ~5.0.2
    expo-local-authentication@13.6.0 - expected version: ~13.4.1
    expo-location@16.3.0 - expected version: ~16.1.0
    expo-splash-screen@0.22.0 - expected version: ~0.20.5
    expo-status-bar@1.7.1 - expected version: ~1.6.0
    expo-store-review@6.6.0 - expected version: ~6.4.0
    react-native@0.73.1 - expected version: 0.72.6
    react-native-gesture-handler@2.14.0 - expected version: ~2.12.0
    react-native-get-random-values@1.10.0 - expected version: ~1.9.0
    react-native-reanimated@3.6.1 - expected version: ~3.3.0
    react-native-safe-area-context@4.8.2 - expected version: 4.6.3
    react-native-screens@3.29.0 - expected version: ~3.22.0
    react-native-svg@14.1.0 - expected version: 13.9.0
    react-native-webview@13.6.4 - expected version: 13.2.2
  Your project may not work correctly until you install the correct versions of the packages.
  Fix with: npx expo install --fix
  Found outdated dependencies

### Reproducible demo or steps to reproduce from a blank project

Update to the latest react native version `"react-native": "0.73.1",` and you will reproduce the error. If I try `npx expo prebuild` with the same cocoapod version we get from EAS then I can reproduce the error.

```
[!] An error occurred while processing the post-install hook of the Podfile.

undefined method `__apply_Xcode_12_5_M1_post_install_workaround' for #<Pod::Podfile:0x00000001086417d0 @defined_in_file=#<Pathname:
```
I tried with an update version of CocoaPod and I'm getting the same error. On internet I found we can just remove the following call 

```

__apply_Xcode_12_5_M1_post_install_workaround(installer)

```

from Expo Podfile

**Comment by szdziedzic** [comment]
Can you send me the URL of your build?

**Comment by santomegonzalo** [comment]
Hi @szdziedzic... any way I can share this with you in private?

**Comment by szdziedzic** [comment]
Yeah, of course, you can for example DM me on Twitter https://twitter.com/szdziedzic. If it doesn't work for you please DM me using Discord, my username is also **szdziedzic**.

**Comment by ErikTromp** [comment]
I am having the same error, locally using Expo 49.0.21 with RN 0.73.2.
Running `eas build --profile development --platform ios` and seeing Runtime version exposdk:49.0.0 on the eas build.

**Comment by szdziedzic** [comment]
Do you have not gitignored `Podfile` in your project?

**Comment by ErikTromp** [comment]
I do not even have a .gitignore in my project.

**Comment by omardiaa** [comment]
I have same issue and here is my [build link](https://expo.dev/accounts/omar_/projects/Chemistry-School/builds/6a5e69f8-ccae-426d-a293-84350ee499a9) 

I don't have a Mac and I cannot test locally (I don't have a Podfile). 

@szdziedzic Could you please check? 🙏🏻

**Comment by omardiaa** [comment]
I opened another issue [here](https://github.com/expo/eas-cli/issues/2194)

**Comment by szdziedzic** [comment]
https://github.com/expo/eas-cli/issues/2194#issuecomment-1904315915
